assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/filter/FactoryRetrieverFilter.java

```diff
@@ -109,6 +114,28 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain filte
                     return;
                 } 
                 requestedFactory = matchedFactories.get(0);
+            } else {
+                // asked for a parameters factory
+
+                // first populate map of parameters
+                Map<String, String> map = new HashMap<>();
+                Enumeration<String> parameterNames = httpReq.getParameterNames();
+                while (parameterNames.hasMoreElements()) {
+                    String parameterName  = parameterNames.nextElement();
+                    map.put(parameterName, httpReq.getParameter(parameterName));
+                }
+
+                // Create URL
+                final String getFactoryUrl = fromUri(apiEndPoint).path(ParametersFactoryService.class)
+                                                                 .path(ParametersFactoryService.class, "getFactory")
+                                                                 .build().toString();
+                // perform call
+                requestedFactory = httpRequestFactory.fromUrl(getFactoryUrl)
+                                                     .setMethod(POST)
+                                                     .setBody(map)
+                                                     .request()
+                                                     .asDto(Factory.class);
+
```

Comment:
And why we didn't check result of the call? if it is 500? We should do  dispatchToErrorPage here as well imho.
Also, for me looks like [this](https://github.com/codenvy/codenvy/blob/71d1b5b03f7668ac7df34926b733d08eefee4d33/assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/filter/FactoryParamsFilter.java#L100-L131) check can be avoided by catching correct answer from Factory service here